pyhf/tensor/jax_backend.py

```diff
@@ -0,0 +1,261 @@
+import jax.numpy as np
+import numpy as onp
+import logging
+from scipy.special import gammaln
+from scipy.stats import norm
+
+log = logging.getLogger(__name__)
+
+
+class jax_backend(object):
+    """jax backend for pyhf"""
+
+    def __init__(self, **kwargs):
+        self.name = 'jax'
+
+    def clip(self, tensor_in, min_value, max_value):
+        """
+        Clips (limits) the tensor values to be within a specified min and max.
+
+        Example:
+
+            >>> import pyhf
+            >>> pyhf.set_backend(pyhf.tensor.numpy_backend())
+            >>> a = pyhf.tensorlib.astensor([-2, -1, 0, 1, 2])
+            >>> pyhf.tensorlib.clip(a, -1, 1)
+            array([-1., -1.,  0.,  1.,  1.])
+
+        Args:
+            tensor_in (`tensor`): The input tensor object
+            min_value (`scalar` or `tensor` or `None`): The minimum value to be cliped to
+            max_value (`scalar` or `tensor` or `None`): The maximum value to be cliped to
+
+        Returns:
+            NumPy ndarray: A clipped `tensor`
+        """
+        return np.clip(tensor_in, min_value, max_value)
+
+    def tolist(self, tensor_in):
+        try:
+            return onp.asarray(tensor_in).tolist()
+        except AttributeError:
+            if isinstance(tensor_in, list):
+                return tensor_in
+            raise
+
+    def outer(self, tensor_in_1, tensor_in_2):
+        tensor_in_1 = self.astensor(tensor_in_1)
+        tensor_in_2 = self.astensor(tensor_in_2)
+        return np.outer(tensor_in_1, tensor_in_2)
+
+    def gather(self, tensor, indices):
+        return tensor[indices]
+
+    def boolean_mask(self, tensor, mask):
+        return tensor[mask]
+
+    def isfinite(self, tensor):
+        return np.ones_like(tensor, dtype=onp.bool)
+        # return np.isfinite(tensor)
```

Comment:
Looks like there are some commented out bits that need to get cleaned up or given further information

Reply:
these were copied over from numpy_backend.py